src/processes/scoring/calc-user-score.js

```diff
@@ -57,11 +57,12 @@ const calcUserScore = async (userDoc) => {
   calcKarmaScore(userDoc._id, user_score);
 
   // checking and sync date_created
-  if (userDoc.created_at === null) {
-    // if created_at is null, set it to created_at in db
+  if (userDoc.created_at === null || userDoc.register_time === null) {
+    // if created_at or register_tim is null, set it to created_at in db
     const user = await UsersFunction.getUserByUserId(userDoc._id);
     if (user) {
       userDoc.created_at = user.created_at;
+      userDoc.register_time = user.created_at;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic here is not entirely correct. If `userDoc.register_time` is null but `userDoc.created_at` is not, the code will still execute and overwrite `userDoc.created_at`. It would be better to handle these two fields separately to avoid unnecessary database calls and overwrites.

```diff
-  if (userDoc.created_at === null || userDoc.register_time === null) {
-    // if created_at or register_tim is null, set it to created_at in db
-     const user = await UsersFunction.getUserByUserId(userDoc._id);
-     if (user) {
-       userDoc.created_at = user.created_at;
-       userDoc.register_time = user.created_at;
-    }
-  }

+  if (userDoc.created_at === null) {
+    const user = await UsersFunction.getUserByUserId(userDoc._id);
+    if (user) {
+      userDoc.created_at = user.created_at;
+    }
+  }
+
+  if (userDoc.register_time === null) {
+    const user = await UsersFunction.getUserByUserId(userDoc._id);
+    if (user) {
+      userDoc.register_time = user.created_at;
+    }
+  }
```
Also, there's a typo in the comment on line 61, it should be `register_time`, not `register_tim`.
```diff
- // if created_at or register_tim is null, set it to created_at in db
+ // if created_at or register_time is null, set it to created_at in db
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->